src/ChunkedDecoder.php

```diff
@@ -139,11 +141,15 @@ public function handleData($data)
                 $this->headerCompleted = false;
                 $this->transferredSize = 0;
                 $this->buffer = (string)substr($this->buffer, 2);
-            } else if ($this->chunkSize === $this->transferredSize && strlen($this->buffer) > 2) {
```

Comment:
The idea behind my comment was to put a return here as well. The reasoning behind this is a) we're done after this point of code and there is no reason to continue running code in this function. And b) is lowers cognitive load on developers reading and working on this code.

In case you're interested in these little code readability optimizations I would suggest https://www.youtube.com/watch?v=GtB5DAfOWMQ if you haven't seen it. If you have my apologies :smile: